server/py/services/api/migrations/versions/d78bd07756c1_fix_alert_activation_partition_.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""fix_alert_activation_partition_expression
+
+Revision ID: d78bd07756c1
+Revises: 57d26493fbff
+Create Date: 2024-12-17 15:29:44.532042
+
+"""
+
+import os
+from datetime import datetime
+
+from alembic import op
+
+import mlrun.common.schemas.alert
+import mlrun.common.schemas.partition
+
+# revision identifiers, used by Alembic.
+revision = "d78bd07756c1"
+down_revision = "57d26493fbff"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    partition_interval = os.getenv("PARTITION_INTERVAL", "YEARWEEK").upper()
+    if partition_interval != "YEARWEEK":
+        # Validate the partition interval
+        if not mlrun.common.schemas.partition.PartitionInterval.is_valid(
+            partition_interval
+        ):
+            raise ValueError(
+                f"Partition interval can only be one of the following: "
+                f"{mlrun.common.schemas.partition.PartitionInterval.valid_intervals()}"
+            )
+
+        # Calculate the date of next partitioning interval
+        now_utc = datetime.utcnow()
+
+        partition_interval = mlrun.common.schemas.partition.PartitionInterval(
+            partition_interval
+        )
+
+        partition_name, partition_value = partition_interval.get_partition_info(
+            now_utc
+        )[0]
+        partition_expression = partition_interval.get_partition_expression(
+            column_name="activation_time"
+        )
+
+        # Construct SQL for partitioning
+        partition_sql = f"""
+            ALTER TABLE alert_activations
+            PARTITION BY RANGE ({partition_expression}) (
+                PARTITION p{partition_name} VALUES LESS THAN ({partition_value})
+            );
+        """
```

Comment:
For new systems (e.g. on the final 1.8.0) the partitioning will already be ok, since the original migration is fixed too.
Maybe we can validate this and skip this migration if it isn't needed.

Reply:
@TomerShor yes, that's absolutely right

okay, then removing the migration